code/modules/power/lighting.dm

```diff
@@ -614,12 +616,32 @@
 	var/rigged = 0		// true if rigged to explode
 	var/brightness_range = 2 //how much light it gives off
 	var/brightness_power = 1
-	var/brightness_color = null
+	var/brightness_color
+	var/time_to_paint = 30  //amount of time it takes to paint a light
+	var/time_to_clean = 40 //amount of time it takes to clean a light
 
 /obj/item/light/ComponentInitialize()
 	. = ..()
 	AddComponent(/datum/component/caltrop, force)
 
+/obj/item/light/attackby(obj/item/O, mob/M)
+	..()
+	//color the light with a spraycan
+	if(istype(O, /obj/item/toy/crayon/spraycan) && do_after(M, time_to_paint, target = src))
+		var/obj/item/toy/crayon/spraycan/S = O
+		brightness_color =  S.colour
+		color = S.colour
+		visible_message("<span class='warning'>[M] paints the [src]!</span>")
+		update()
+		M.do_attack_animation(src)
+	//clean the light
+	else if(istype(O, /obj/item/soap) && do_after(M, time_to_clean, target = src))
+		brightness_color = null
```

Comment:
This won't break? Since it used to always have a value before

Reply:
Good point, how some have a set color like the small bulbs. I will see what I can do

Reply:
I think it makes more sense to just remove any color something has after cleaning it since I'd have to check for a certain color and it's pretty much the same anyways - light peach and white.

Nothing breaks except it makes it white